### ISSUE TITLE
problems when generating optimized models 

### ISSUE THREAD

**Issue by AouatifZ** [hailo-ai/hailo_model_zoo#63]
Hi all, 

I'm new to hailo optimization, 

I tried to prepare the necessary environment to compile my custom model,  (yaml file, tfrecord dataset.....)
first I converted my model to model.har (parse step) 
then I tried to use this output in the second optimize step but an error appeared, 
![errorgithub](https://github.com/hailo-ai/hailo_model_zoo/assets/111449259/6ab2d183-82f8-46c0-bcdb-71937d766438)

I don't know what the problem is, or how to solve it. 

any help or suggestion please

**Comment by nina-vilela** [comment]
Hi @AouatifZ 

There seems to be an issue with the model script path in the yaml being used. 

In our standard [yolov5l yaml](https://github.com/hailo-ai/hailo_model_zoo/blob/master/hailo_model_zoo/cfg/networks/yolov5l.yaml), you can see that the model script used is [this one](https://github.com/hailo-ai/hailo_model_zoo/blob/master/hailo_model_zoo/cfg/alls/generic/yolov5l.alls). Here are the lines in the yaml that define which model script to use:

`paths:`
 ` alls_script: yolov5l.alls`

Make sure that these lines are part of the yaml you are using.

**Comment by AouatifZ** [comment]
thank you for your information and your reply, 

in my yaml file, I tried to define only the path of my custom onnx model and I removed the other two parameters of url and alls_scripts since these two parameters defined in the yaml standart of yolov5 dedicated to load the pre-trained model

aren't they 
![yamlcustom](https://github.com/hailo-ai/hailo_model_zoo/assets/111449259/196d2557-a6ed-421c-a435-56b34511ea9d)

**Comment by AouatifZ** [comment]
Thanks @nina-vilela 

I tried to do this, here is the yolov5.alls file
![l1](https://github.com/hailo-ai/hailo_model_zoo/assets/111449259/bcb02749-55b4-4b5d-957b-d2b840c0789f)

and here is the output of my command

hailomz optimize --ckpt best.onnx --yaml yolov5l.yaml  --har yolov5l.har --calib-path Aouatif_hailo_test/valdata/  --performance

![l2](https://github.com/hailo-ai/hailo_model_zoo/assets/111449259/5cccef6d-3bdf-41d0-8228-f5790ff6e41a)


how can I find out the names of the layers corresponding to the loss

**Comment by nina-vilela** [comment]
@AouatifZ 
If you only re-trained the model, you can use a visualizing tool, such as Netron, to visualize both the original yolov5l and the custom model. This way you can find the selected end-node layers on your model. 

If the model contains modifications, then the end-node layers could depend on what these modifications were.

**Comment by AouatifZ** [comment]
@nina-vilela 

after viewing my ONNX format model  on Netron, I found that these layers are responsible for producing the bounding box predictions, object scores, and class scores for the detected objects.

I've modified my configuration but the problem remains in the post quantization optimization part. 

![l4](https://github.com/hailo-ai/hailo_model_zoo/assets/111449259/be8126f0-b3be-4d92-aede-b69d7fde8a0e)

I do not know the exact problem that finds Hailo during the optimization of my model.

**Comment by nina-vilela** [comment]
@AouatifZ 

The loss layers should be passed with the layer names after the parsing. You can find them by visualizing the parsed har, which can also be done with Netron.

**Comment by AouatifZ** [comment]
@nina-vilela 
please can you give me more information and more details
